lib/call.ts

```diff
@@ -101,6 +101,22 @@ export function call<T, TArgs extends unknown[] = []>(
   };
 }
 
+/**
+ * It can be used to treat a promise as an operation. This function
+ * is a replacement to the v3 deprecated `call(promise)` function form.
+ *
+ * @example
+ * ```js
+ * let response = yield* until(fetch('https://google.com'));
+ * ```
+ * @template {T}
+ * @param promise
+ * @returns {Operation<T>}
+ */
+export function until<T>(promise: PromiseLike<T>): Operation<T> {
+  return call(async () => await promise);
+}
+
```

Comment:
Let's implement `until()` directly with an `action()`. There is no need to go through the full `call()` logic every time we want to await a promise.

Also, `until()` is its own API independent from `call()` that belongs in its own file with its own test. That way, `call()` can depend on `until()` and not the other way around.